tripal/tripal
# Tripal 4 sequence coordinate field

Field Sequence Coordinates




#
### Issue # 1414

### Tripal Version:  4
## Description : 

This is required for Sequence Coordinates field to run in Tripal 4. Since the YML file for the field has been updated, this field should be active for the Gene page.  The Tripal-3 version of this field can be viewed here: 
https://github.com/tripal/tripal/tree/7.x-3.x/tripal_chado/includes/TripalFields/data__sequence_coordinates

Two formatters are present one for linear view of the coordinates and other for a tabular view.


## Testing?



1. Build docker image with the code in respective directories:
```
tripal_chado/config/install/tripal.tripalfield_collection.genomic_chado.yml
tripal_chado/src/Plugin/Field/FieldType/ChadoSequenceCoordinatesDefault.php
tripal_chado/src/Plugin/Field/FieldFormatter/ChadoSequenceCoordinatesFormatterDefault.php
tripal_chado/src/Plugin/Field/FieldFormatter/ChadoSequenceCoordinatesFormatterTable.php
tripal_chado/src/Plugin/Field/FieldWidget/ChadoSequenceCoordinatesWidgetDefault.php

```
2. Build and run docker image as required and verify Tripal can be accessed from its URL.

3.  Access docker container using branch name : tv4g1-issue1414-data__sequence_coordinates
```
cntr_br_name=tv4g1-issue1414-data__sequence_coordinates
docker exec -it $cntr_br_name /bin/bash
```

4. Change max_filesize value in settings file : `/usr/local/etc/php/php.ini` to a high value say 12MB.
`service apache2 restart` and exit from container. (This step is not required to just publish gene imported via Tripal -> Data Loaders -> Chado GFF3 File Loader )

5.  Log into Drupal/Tripal GUI and create Citrus Organism Tripal Content 
Ref: https://tripal.readthedocs.io/en/latest/user_guide/example_genomics/organisms.html

6. Create Analysis content :
Ref: https://tripal.readthedocs.io/en/latest/user_guide/example_genomics/analyses.html

7.  Structure -> Tripal Content Types -> Import type collection -> Genomic Content Types for viewing Gene content.

8. Load Feature Data (gene and sequence information)
Ref: https://tripal.readthedocs.io/en/latest/user_guide/example_genomics/genomes_genes.html
(Tripal -> Data Loaders -> Chado GFF3 File Loader )

9. Publish the gene (Content -> Tripal Content -> +Publish Tripal Content -> Chado Storage, Content Type: Gene -> Publish)

10. Click on the Tripal Content -> gene that was just created.

11. If the field is working properly, the gene entered via the GUI would be viewable and editable with the relevant coordinate information under Sequence Coordinates.  

12.  Structure -> Tripal Content Types -> Gene -> Edit -> Manage Display -> Change Sequence Coordinates formatter to "Chado sequence coordinates table formatter" and see coordinates being displayed as a table.
